packages/mui-joy/src/Badge/Badge.tsx

```diff
@@ -216,7 +216,7 @@ const Badge = React.forwardRef(function Badge(inProps, ref) {
         Badge: BadgeBadge,
         ...components,
       }}
-      className={clsx(className, classes.root, componentsProps.root?.className)}
+      className={clsx(className, classes.root)}
```

Comment:
Handling classes from componentsProps is done in useSlotProps, there's no need to do it here.